cupy/core/_routines_indexing.pyx

```diff
@@ -49,6 +49,16 @@ cdef _ndarray_setitem(ndarray self, slices, value):
 
 
 cdef tuple _ndarray_nonzero(ndarray self):
+    cdef int ndim
+    ndim = max(<int>self._shape.size(), 1)
```

Comment:
Think it's just this

```suggestion
    ndim = self.ndim
```

Reply:
Alternatively this could just be `len(dst)` since this should already be baked into the result.